### PR TITLE
spark etl support datalake and add scala-2.12 module

### PR DESCRIPTION
**Pull request by ChengJie1053**
spark etl support datalake and add scala-2.12 module

### CODE REVIEW

File: linkis-engineconn-plugins/spark/scala-2.12/pom.xml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.linkis</groupId>
+    <artifactId>linkis</artifactId>
+    <version>${revision}</version>
+    <relativePath>../../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>linkis-engineplugin-scala-2.12</artifactId>
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/scala-2.12/pom.xml:27]: linkis-engineplugin-spark-scala-2.12?

File: linkis-engineconn-plugins/spark/scala-2.12/src/main/assembly/distribution.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <format>zip</format>
+    </formats>
+    <includeBaseDirectory>true</includeBaseDirectory>
+    <baseDirectory>scala</baseDirectory>
+
+    <dependencySets>
+        <dependencySet>
+            <!-- Enable access to all projects in the current multimodule build! <useAllReactorProjects>true</useAllReactorProjects> -->
+            <!-- Now, select which projects to include in this module-set. -->
+            <outputDirectory>/dist/2.12/lib</outputDirectory>
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/scala-2.12/src/main/assembly/distribution.xml:33]: The directory should be consistent with Spark EC?

File: linkis-engineconn-plugins/spark/scala-2.12/src/test/scala/org/apache/linkis/engineplugin/spark/datacalc/TestDeltaCalc.scala [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  val writeConfigJson =
+    """
+      |{
+      |    "sources": [
+      |        {
+      |            "name": "file",
+      |            "type": "source",
+      |            "config": {
+      |                "resultTable": "T1654611700631",
+      |                "path": "file:///Users/chengjie/cjtest/newfile.csv",
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/scala-2.12/src/test/scala/org/apache/linkis/engineplugin/spark/datacalc/TestDeltaCalc.scala:58]: use test resource

Comment (ChengJie1053) [reply linkis-engineconn-plugins/spark/scala-2.12/src/test/scala/org/apache/linkis/engineplugin/spark/datacalc/TestDeltaCalc.scala:58]: Ok, thanks for reviewing the code

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**